### PR TITLE
Set default serverHost value when none is present.

### PR DESCRIPTION
**Pull request by yanscalyr**
Also adds a serverHost value of the actual logstash node to status messages.

### CODE REVIEW

File: lib/logstash/outputs/scalyr.rb [hunk 1]
```diff
@@ -361,16 +361,25 @@ def build_multi_event_request_array(logstash_events)
 
       # Rename user-specified serverHost field -> 'serverHost'
       rename.call(@serverhost_field, 'serverHost')
-      record.delete(@serverhost_field)
+      if @serverhost_field != 'serverHost'
```
Comment (oliverhsu77) [inline lib/logstash/outputs/scalyr.rb:364]: `rename` lambda is already deleting the field on line 345.  Why is this needed?

File: lib/logstash/outputs/scalyr.rb [hunk 2]
```diff
@@ -562,6 +571,7 @@ def send_status
 
     if !@last_status_transmit_time
       status_event[:attrs]['message'] = "Started Scalyr LogStash output plugin."
+      status_event[:attrs]['serverHost'] = Socket.gethostname
```
Comment (oliverhsu77) [inline lib/logstash/outputs/scalyr.rb:574]: Not sure the cost of this call.  Should we call `Socket.gethostname` once and cache the value instead of every time we send the status?

File: lib/logstash/outputs/scalyr.rb [hunk 3]
```diff
@@ -361,16 +361,25 @@ def build_multi_event_request_array(logstash_events)
...
+        record.delete(@logfile_field)
+      end
       # Set logfile field if empty and serverHost is supplied
       if record['logfile'].to_s.empty? and serverHost
         record['logfile'] = "/logstash/#{serverHost}"
       end
 
+      # Set a default if no serverHost value is present.
+      if serverHost.nil?
+        record['serverHost'] = "Logstash"
```
Comment (oliverhsu77) [inline lib/logstash/outputs/scalyr.rb:380]: Should we default to `Logstash` or the actual Logstash hostname from `Socket.gethostname`.

Comment (yanscalyr) [reply lib/logstash/outputs/scalyr.rb:380]: Here we are attaching it to the events from unknown sources, which are likely not coming from the same serverhost as this logstash instance, thats why we want a simple default like that. We use `Socket.gethostname` in the status messages for the instance itself.
